networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -91,6 +92,19 @@
     this.eth2RpcPingInterval = eth2RpcPingInterval;
     this.eth2RpcOutstandingPingThreshold = eth2RpcOutstandingPingThreshold;
     this.eth2StatusUpdateInterval = eth2StatusUpdateInterval;
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () ->
+                    connectedPeerMap
+                        .values()
+                        .forEach(
+                            peer ->
+                                peer.sendGoodbye(GoodbyeMessage.REASON_CLIENT_SHUT_DOWN)
+                                    .finish(
+                                        err ->
+                                            LOG.debug(
+                                                "Unable to send goodbye to peer {}", peer, err)))));
```

Comment:
We should just tie this into the existing stop functionality rather than adding another shutdown hook.
Also the write is async so we need to actually wait for it to complete (as a batch, don't wait sequentially for each peer).

Would be best to use the disconnectCleanly method as well so the peer is marked as disconnected and we don't send any other messages after disconnect.

Reply:
Done.